platforms/core-runtime/daemon-protocol/src/main/java/org/gradle/launcher/daemon/server/api/DaemonState.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.launcher.daemon.server.api;
+
+public enum DaemonState {Idle, Busy, Canceled, StopRequested, Stopped, ForceStopped, Broken}
```

Comment:
💅 Should we put those one per line?

Most values are self-explanatory, but I would personally love to have a javadoc at least on the `Broken` state, because it's unclear what causes this state and what it means when the daemon is in this state.

It would be ideal if we would have a diagram for the state transitions. But I would at least start something non-trivial like whether the daemon can return to `Idle` from any other states.

Reply:
+1

Reply:
The goal for this PR is simply to move things around, not improve them.